src/borrow_tracker/tree_borrows/diagnostics.rs

```diff
@@ -28,7 +28,7 @@ impl fmt::Display for AccessCause {
             Self::Explicit(kind) => write!(f, "{kind}"),
             Self::Reborrow => write!(f, "reborrow"),
             Self::Dealloc => write!(f, "deallocation"),
-            Self::FnExit => write!(f, "protector release"),
+            Self::FnExit(kind) => write!(f, "protector release ({kind})"),
```

Comment:
"protector release read" and "protector release write" would be a bit more concise than "protector release (read access)".
```suggestion
            Self::FnExit(AccessKind::Read) => write!(f, "protector release read"),
            Self::FnExit(AccessKind::Write) => write!(f, "protector release write"),
```
But also we don't seem to have any test that even uses this string? That's a bit unfortunate. Oh well.

Reply:
Indeed, I could not figure out where this is used so I didn't think about how it'd look in context.

Reply:
It seems the only place it is used is [here](https://github.com/rust-lang/miri/blob/81f85ec72e1a98c244838a2a3d5adc5a4a75276d/src/borrow_tracker/tree_borrows/diagnostics.rs#L306)

Reply:
I think this can never happen..

Reply:
Oh, really? Why?

Reply:
Ah I guess the protectors ensure these accesses would be allowed, that's the entire point.

Reply:
well, you would need a protected initialized tag that's a cousin to another protected initialized tag. This means none of the tags can be `Active` since it's impossible to create that situation. It is possible for them to be `Frozen` or `Reserved`, and then they survive the foreign read emitted by the protector end semantics.